### PR TITLE
🦺 Consent DAS error handling for Clinician Invite flow (#275)

### PR DESCRIPTION
**Pull request by alyssadsouza**
## Description of Changes

Implemented complete error handling, validation, and logging for the `POST /invites` router and `createClinicianInvite` service in Consent DAS using `consent-api` and `pi-das` as an example.

### Consent DAS
- Added `express-request-validation` from shared packages to use `withRequestValidation` to validate the endpoint's request body
- Updated the endpoint's HTTP Responses to use the appropriate `ErrorResponse`
- Added a 400 RequestValidationError and 409 ConflictError to the endpoint responses, updated the JSDocs
- Updated the return type of the `createClinicianInvite` service to `Result` so it returns appropriate `Success` and `Failure` responses to the router
- Updated the Prisma create request so it returns a failure response if an error occurred
>**Note:** The endpoint request body validation was updated to use the Zod schema `ConsentClinicianInvite` so it now expects `id` as a request body field instead of `clinicianId`

### Special Instructions
Before running these changes, you will need to add `express-request-validation` to Consent DAS:
```
pnpm i
```

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [ ] Added copyrights to new files

### CODE REVIEW

File: apps/consent-das/src/service/create.ts [hunk 1]
```diff
@@ -81,40 +88,42 @@ export const createParticipantResponse = async ({
...
+		.create({
+			data: inviteRequest,
+		})
+		.then((invite) => success(invite))
+		.catch((error) => {
+			if (error instanceof PrismaClientKnownRequestError) {
+				if (error.code === 'P2002') {
+					// Prisma error code P2002 indicates "Unique constraint failed"
+					// See docs: https://www.prisma.io/docs/reference/api-reference/error-reference#p2002
+					logger.info('POST /invites', 'Unique constraint failed creating invite.', error.message);
```
Comment (anncatton) [inline apps/consent-das/src/service/create.ts:110]: just noticed this log uses `info` level - curious why not `error`? (same as in the pi-das pr but i missed it there)

Also looking at the prisma error type i think the log message `'Unique constraint failed creating invite.'` will be repeated in the error message, might be better to use language similar to the `failure` message.

**Review by anncatton — COMMENTED** [review]
Just a minor comment on the log statements, otherwise looks great!

**Review by anncatton — APPROVED** [review]
wooooo thank ya 🎉

**Review by mistryrn — APPROVED** [review]
Thanks for getting those changes in so quick! 🚀

**Pull request merged**